diff --git a/std/fs.zig b/std/fs.zig
index 64a909d..8567230 100644
--- a/std/fs.zig
+++ b/std/fs.zig
@@ -50,7 +50,7 @@ pub const GetAppDataDirError = @import("fs/get_app_data_dir.zig").GetAppDataDirE
 /// * On other platforms, `[]u8` file paths are opaque sequences of bytes with
 ///   no particular encoding.
 pub const max_path_bytes = switch (native_os) {
-    .linux, .macos, .ios, .freebsd, .openbsd, .netbsd, .dragonfly, .haiku, .solaris, .illumos, .plan9, .emscripten, .wasi, .serenity => posix.PATH_MAX,
+    .linux, .macos, .ios, .tvos, .watchos, .visionos, .freebsd, .openbsd, .netbsd, .dragonfly, .haiku, .solaris, .illumos, .plan9, .emscripten, .wasi, .serenity => posix.PATH_MAX,
     // Each WTF-16LE code unit may be expanded to 3 WTF-8 bytes.
     // If it would require 4 WTF-8 bytes, then there would be a surrogate
     // pair in the WTF-16LE, and we (over)account 3 bytes for it that way.
@@ -71,7 +71,7 @@ pub const max_path_bytes = switch (native_os) {
 /// On WASI, file name components are encoded as valid UTF-8.
 /// On other platforms, `[]u8` components are an opaque sequence of bytes with no particular encoding.
 pub const max_name_bytes = switch (native_os) {
-    .linux, .macos, .ios, .freebsd, .openbsd, .netbsd, .dragonfly, .solaris, .illumos, .serenity => posix.NAME_MAX,
+    .linux, .macos, .ios, .tvos, .watchos, .visionos, .freebsd, .openbsd, .netbsd, .dragonfly, .solaris, .illumos, .serenity => posix.NAME_MAX,
     // Haiku's NAME_MAX includes the null terminator, so subtract one.
     .haiku => posix.NAME_MAX - 1,
     // Each WTF-16LE character may be expanded to 3 WTF-8 bytes.
diff --git a/std/fs/Dir.zig b/std/fs/Dir.zig
index 768efde..2ff56aa 100644
--- a/std/fs/Dir.zig
+++ b/std/fs/Dir.zig
@@ -39,7 +39,7 @@ const IteratorError = error{
 } || posix.UnexpectedError;
 
 pub const Iterator = switch (native_os) {
-    .macos, .ios, .freebsd, .netbsd, .dragonfly, .openbsd, .solaris, .illumos => struct {
+    .macos, .ios, .tvos, .watchos, .visionos, .freebsd, .netbsd, .dragonfly, .openbsd, .solaris, .illumos => struct {
         dir: Dir,
         seek: i64,
         buf: [1024]u8 align(@alignOf(posix.system.dirent)),
@@ -55,7 +55,7 @@ pub const Iterator = switch (native_os) {
         /// with subsequent calls to `next`, as well as when this `Dir` is deinitialized.
         pub fn next(self: *Self) Error!?Entry {
             switch (native_os) {
-                .macos, .ios => return self.nextDarwin(),
+                .macos, .ios, .tvos, .watchos, .visionos => return self.nextDarwin(),
                 .freebsd, .netbsd, .dragonfly, .openbsd => return self.nextBsd(),
                 .solaris, .illumos => return self.nextSolaris(),
                 else => @compileError("unimplemented"),
@@ -615,6 +615,9 @@ fn iterateImpl(self: Dir, first_iter_start_value: bool) Iterator {
     switch (native_os) {
         .macos,
         .ios,
+        .tvos,
+        .watchos,
+        .visionos,
         .freebsd,
         .netbsd,
         .dragonfly,
@@ -1678,7 +1681,7 @@ pub fn deleteFileZ(self: Dir, sub_path_c: [*:0]const u8) DeleteFileError!void {
         error.AccessDenied, error.PermissionDenied => |e| switch (native_os) {
             // non-Linux POSIX systems return permission errors when trying to delete a
             // directory, so we need to handle that case specifically and translate the error
-            .macos, .ios, .freebsd, .netbsd, .dragonfly, .openbsd, .solaris, .illumos => {
+            .macos, .ios, .tvos, .watchos, .visionos, .freebsd, .netbsd, .dragonfly, .openbsd, .solaris, .illumos => {
                 // Don't follow symlinks to match unlinkat (which acts on symlinks rather than follows them)
                 const fstat = posix.fstatatZ(self.fd, sub_path_c, posix.AT.SYMLINK_NOFOLLOW) catch return e;
                 const is_dir = fstat.mode & posix.S.IFMT == posix.S.IFDIR;
diff --git a/std/process/Child.zig b/std/process/Child.zig
index 39dab9f..7d555e4 100644
--- a/std/process/Child.zig
+++ b/std/process/Child.zig
@@ -134,7 +134,7 @@ pub const ResourceUsageStatistics = struct {
                     return null;
                 }
             },
-            .macos, .ios => {
+            .macos, .ios, .tvos, .watchos, .visionos => {
                 if (rus.rusage) |ru| {
                     // Darwin oddly reports in bytes instead of kilobytes.
                     return @as(usize, @intCast(ru.maxrss));
@@ -147,7 +147,7 @@ pub const ResourceUsageStatistics = struct {
     }
 
     const rusage_init = switch (native_os) {
-        .linux, .macos, .ios => @as(?posix.rusage, null),
+        .linux, .macos, .ios, .tvos, .watchos, .visionos => @as(?posix.rusage, null),
         .windows => @as(?windows.VM_COUNTERS, null),
         else => {},
     };
@@ -477,7 +477,7 @@ fn waitUnwrappedPosix(self: *ChildProcess) void {
     const res: posix.WaitPidResult = res: {
         if (self.request_resource_usage_statistics) {
             switch (native_os) {
-                .linux, .macos, .ios => {
+                .linux, .macos, .ios, .tvos, .watchos, .visionos => {
                     var ru: posix.rusage = undefined;
                     const res = posix.wait4(self.id, 0, &ru);
                     self.resource_usage_statistics.rusage = ru;
diff --git a/std/debug/Dwarf/abi.zig b/std/debug/Dwarf/abi.zig
index b8b644e..f1b422f 100644
--- a/std/debug/Dwarf/abi.zig
+++ b/std/debug/Dwarf/abi.zig
@@ -245,7 +245,7 @@ pub fn regBytes(
                 // TODO: Extract xmm state from sc_fpstate?
                 else => error.InvalidRegister,
             },
-            .macos, .ios => switch (reg_number) {
+            .macos, .ios, .tvos, .watchos, .visionos => switch (reg_number) {
                 0 => mem.asBytes(&ucontext_ptr.mcontext.ss.rax),
                 1 => mem.asBytes(&ucontext_ptr.mcontext.ss.rdx),
                 2 => mem.asBytes(&ucontext_ptr.mcontext.ss.rcx),
@@ -291,7 +291,7 @@ pub fn regBytes(
             else => error.UnimplementedOs,
         },
         .aarch64, .aarch64_be => switch (builtin.os.tag) {
-            .macos, .ios, .watchos => switch (reg_number) {
+            .macos, .ios, .tvos, .watchos, .visionos => switch (reg_number) {
                 0...28 => mem.asBytes(&ucontext_ptr.mcontext.ss.regs[reg_number]),
                 29 => mem.asBytes(&ucontext_ptr.mcontext.ss.fp),
                 30 => mem.asBytes(&ucontext_ptr.mcontext.ss.lr),
diff --git a/std/debug/SelfInfo.zig b/std/debug/SelfInfo.zig
index 3f1514a..80e8910 100644
--- a/std/debug/SelfInfo.zig
+++ b/std/debug/SelfInfo.zig
@@ -1812,7 +1812,7 @@ pub fn supportsUnwinding(target: *const std.Target) bool {
             else => false,
         },
         .x86_64 => switch (target.os.tag) {
-            .linux, .netbsd, .freebsd, .openbsd, .macos, .ios, .solaris, .illumos => true,
+            .linux, .netbsd, .freebsd, .openbsd, .macos, .ios, .tvos, .watchos, .visionos, .solaris, .illumos => true,
             else => false,
         },
         .arm, .armeb, .thumb, .thumbeb => switch (target.os.tag) {
@@ -1820,7 +1820,7 @@ pub fn supportsUnwinding(target: *const std.Target) bool {
             else => false,
         },
         .aarch64, .aarch64_be => switch (target.os.tag) {
-            .linux, .netbsd, .freebsd, .macos, .ios => true,
+            .linux, .netbsd, .freebsd, .macos, .ios, .tvos, .watchos, .visionos => true,
             else => false,
         },
         // Unwinding is possible on other targets but this implementation does
